src/runtime/pickle_runner.ts

```diff
@@ -395,9 +430,69 @@ export default class PickleRunner {
         },
       })
     }
-    return await this.invokeStep(
-      testStep.pickleStep,
-      testStep.stepDefinitions[0]
+
+    let cumulatedStepResult
+    const beforeStepHooksResult = await this.runStepHooks(
+      this.getBeforeStepHookDefinitions(),
+      cumulatedStepResult
+    )
+    cumulatedStepResult = beforeStepHooksResult
+
+    if (beforeStepHooksResult.status !== Status.FAILED) {
+      const stepResult = await this.invokeStep(
+        testStep.pickleStep,
+        testStep.stepDefinitions[0]
+      )
+      if (stepResult !== undefined) {
+        cumulatedStepResult = stepResult
+        cumulatedStepResult.duration = addDurations(
+          cumulatedStepResult.duration,
+          beforeStepHooksResult.duration
+        )
+        const afterStepHooksResult = await this.runStepHooks(
+          this.getAfterStepHookDefinitions(),
+          cumulatedStepResult
+        )
+        cumulatedStepResult.duration = addDurations(
+          cumulatedStepResult.duration,
+          afterStepHooksResult.duration
+        )
+      }
+    }
+    return cumulatedStepResult
+  }
+
+  async runStepHooks(
+    stepHooks: TestStepHookDefinition[],
+    stepResult: messages.TestStepFinished.ITestStepResult
+  ): Promise<messages.TestStepFinished.ITestStepResult> {
```

Comment:
Lets return the array of results here instead of a single result, shouldn't need any accumulation